airbyte_cdk/entrypoint.py

```diff
@@ -225,7 +241,7 @@ def discover(
         self, source_spec: ConnectorSpecification, config: TConfig
     ) -> Iterable[AirbyteMessage]:
         self.set_up_secret_filter(config, source_spec.connectionSpecification)
-        if self.source.check_config_against_spec:
+        if not hasattr(self.source, "check_config_during_discover") or not self.source.check_config_during_discover:
```

Comment:
Push the class member into the connector base class, adjacent to the existing flag. Then you don't have to wonder if it exists as a member.